docs/blog/2025-01-06-2024-retro.md

```diff
@@ -0,0 +1,118 @@
+
+Looking ahead to 2025, we plan to make the CLI even more accessible by providing precompiled binaries for download, eliminating the need for git clone during installation.
+
+Users can now quickly clone the repository and build an iOS or Android project in just three commands—`mopro init`, `mopro build`, and `mopro create`—all within 3 minutes! For detailed instructions, check out the [Getting Started](/docs/getting-started) section.
+
+### Enabling Multi-Platform Support
+
+In addition to Swift for iOS and Kotlin for Android, we’ve now created templates for cross-platform frameworks like React Native and Flutter. The CLI has been updated to support these platforms, and the documentation has been refreshed to reflect these enhancements.
+
+Please refer to the following resources:
+
+-   [React Native Setup](/docs/setup/react-native-setup)
+-   [React Native App](https://github.com/zkmopro/react-native-app)
+-   [Flutter Setup](/docs/setup/flutter-setup)
+-   [Flutter App](https://github.com/zkmopro/flutter-app)
+
+Additionally, Mopro now supports WASM for web browsers. We provide wasm-bindgen for the Halo2 prover, enabling developers to use the Mopro CLI to generate website templates with bindings. This significantly reduces the time spent navigating the outdated Halo2 tutorial available at [Using halo2 in WASM](https://zcash.github.io/halo2/user/wasm-port.html) (It was authored in 2022).
+
+Please refer to the following resources to learn how to use Mopro for building WASM applications for web browsers:
+
+-   [WASM Setup](/docs/setup/web-wasm-setup)
+
+### Expanding Compatibility with General Rust Functions
+
+We realized that generating and verifying proofs alone isn’t sufficient for application developers. To address this, we made the Mopro template compatible with any Rust crate or function, allowing developers to customize the UDL file to suit their specific needs.
+
+For instance, if a developer needs a Poseidon hash function but neither Swift nor Kotlin provides a Poseidon hash library, they can integrate a Rust Poseidon crate. First, they define the function API in Rust, such as:
+
+```rust
+pub fn poseidon(input: Vec<u8>) -> Vec<u8>{
+    // Poseidon hash implementation
+}
+```
+
+Next, they define the corresponding API in the UDL file. Please refer to [uniffi-rs](https://mozilla.github.io/uniffi-rs/latest/udl_file_spec.html) to learn more about the UDL file format.
+
+For example:
+
+```js
+namespace mopro {
+    // functions...
+    bytes poseidon(bytes input);
+}
+```
+
+By running `mopro build` again, the developer can generate Swift and/or Kotlin bindings for the Poseidon hash function. They can then easily call the function in Swift or Kotlin like this:
+
+```swift
+let hash = poseidon(input: input)
+```
+
+or in kotlin
+
+```kotlin
+val hash = poseidon(input)
+```
+
+Additionally, this approach is compatible with WASM for browsers. You can define a function in Rust as follows:
+
+```rust
+use serde_wasm_bindgen::to_value;
```

Comment:
It should be added this too
```
use wasm_bindgen::prelude::JsValue;
```

Reply:
[6755847](https://github.com/zkmopro/mopro/pull/288/commits/675584747e16b857c3f8e6d710c339e2faa7005e)